test/test.c

```diff
@@ -4969,9 +4969,11 @@ _recvTestString(natsConnection *nc, natsSubscription *sub, natsMsg *msg,
         }
         case 4:
         {
-            arg->status = natsConnection_PublishString(nc,
+            arg->status = natsConnection_Publish(nc,
                                                        natsMsg_GetReply(msg),
-                                                       arg->string);
+                                                       natsMsg_GetData(msg),
```

Comment:
This has to be `arg->string` since the requestor in _sendRequest() checks that the reply content is as expected (arg->string).